src/apyfloat.cc

```diff
@@ -1003,29 +1003,25 @@ APyFloat APyFloat::operator/(const APyFloat& y) const
         new_exp--;
     }
 
+    // Check limits
+    if (new_exp >= res.max_exponent()) {
+        return res.construct_inf();
+    }
+
     // Handle subnormal case
     if (new_exp <= 0) {
         apy_man_res >>= std::abs(new_exp) + 1;
         new_exp = 0;
     }
 
-    // Quantize mantissa
+    // Quantize mantissa. This will never create carry.
     APyFloat::quantize_apymantissa(apy_man_res, res.sign, res.man_bits, quantization);
 
-    // Carry from quantization
-    if (apy_man_res.greater_than_equal_two()) {
-        new_exp++;
-        apy_man_res >>= 1;
-    }
-
-    // Check limits
-    if (new_exp >= res.max_exponent()) {
-        return res.construct_inf();
-    }
-
-    if (apy_man_res.greater_than_equal_one()) { // Remove leading one
+    // Remove leading one
+    if (apy_man_res.greater_than_equal_one()) {
         apy_man_res = apy_man_res - fx_one;
     }
+
     res.man = (man_t)(apy_man_res << res.man_bits).to_double();
```

Comment:
This will create a conflict with my recently merged PR. In general, inline shifting is crucial for good performance of APyFixed (which is added there).